src/EarnerManager.sol

```diff
@@ -0,0 +1,248 @@
+    bytes32 public constant EARNERS_LIST_NAME = "earners";
+
+    /// @inheritdoc IEarnerManager
+    bytes32 public constant MIGRATOR_KEY_PREFIX = "em_migrator_v1";
+
+    /// @inheritdoc IEarnerManager
+    address public immutable registrar;
+
+    /// @inheritdoc IEarnerManager
+    address public immutable migrationAdmin;
+
+    /// @dev Mapping of account to earner details.
+    mapping(address account => EarnerDetails earnerDetails) internal _earnerDetails;
+
+    /* ============ Modifiers ============ */
+
+    modifier onlyAdmin() {
+        _revertIfNotAdmin();
+        _;
+    }
+
+    /* ============ Constructor ============ */
+
+    /**
+     * @dev   Constructs the contract.
+     * @param registrar_      The address of a Registrar contract.
+     * @param migrationAdmin_ The address of a migration admin.
+     */
+    constructor(address registrar_, address migrationAdmin_) {
+        if ((registrar = registrar_) == address(0)) revert ZeroRegistrar();
+        if ((migrationAdmin = migrationAdmin_) == address(0)) revert ZeroMigrationAdmin();
+    }
+
+    /* ============ Interactive Functions ============ */
+
+    /// @inheritdoc IEarnerManager
+    function setEarnerDetails(address account_, bool status_, uint16 feeRate_) external onlyAdmin {
+        if (isEarnersListIgnored()) revert EarnersListIgnored();
+
+        _setDetails(account_, status_, msg.sender, feeRate_);
+    }
+
+    /// @inheritdoc IEarnerManager
+    function setEarnerDetails(
+        address[] calldata accounts_,
+        bool[] calldata statuses_,
+        uint16[] calldata feeRates_
+    ) external onlyAdmin {
+        if (accounts_.length == 0) revert ArrayLengthZero();
+        if (accounts_.length != statuses_.length) revert ArrayLengthMismatch();
+        if (accounts_.length != feeRates_.length) revert ArrayLengthMismatch();
+        if (isEarnersListIgnored()) revert EarnersListIgnored();
+
+        for (uint256 index_; index_ < accounts_.length; ++index_) {
+            // NOTE: The `isAdmin` check in `_setDetails` will make this costly to re-set details for multiple accounts
+            //       that have already been set by the same admin, due to the redundant queries to the registrar.
+            //       Consider transient storage in `isAdmin` to memoize admins.
+            _setDetails(accounts_[index_], statuses_[index_], msg.sender, feeRates_[index_]);
+        }
+    }
+
+    /* ============ Temporary Admin Migration ============ */
+
+    /// @inheritdoc IEarnerManager
+    function migrate(address migrator_) external {
+        if (msg.sender != migrationAdmin) revert UnauthorizedMigration();
+
+        _migrate(migrator_);
+    }
+
+    /* ============ View/Pure Functions ============ */
+
+    /// @inheritdoc IEarnerManager
+    function earnerStatusFor(address account_) external view returns (bool status_) {
+        return isEarnersListIgnored() || isInEarnersList(account_) || _isValidAdmin(_earnerDetails[account_].admin);
```

Comment:
reading will be simpler with more explicit function name

`isInSecondaryEarnersList`